### PR TITLE
Generic issues

### PR DESCRIPTION
**Pull request by amithkanand**
This PR takes care of issues given below:
Issue #82 
Issue #39 
Issue #154 
Issue #153 
Issue #152 
Issue #155 
Issue #156 

This PR also removes all `BCPC` specific roles.

### CODE REVIEW

**Comment by cbaenziger** [comment]
I think you will want to pull the zabbix-agent out of [`build_bins.sh`](https://github.com/amithkanand/chef-bach/blob/739e7f0234098bf63006cbb2151f0abda7443c70/build_bins.sh#L172-L192).

File: cookbooks/bcpc-hadoop/templates/default/hdp_hdfs-site.xml.erb [hunk 1]
```diff
@@ -15,6 +15,11 @@
   </property>
 
   <property>
+    <name>dfs.datanode.balance.bandwidthPerSec</name>
```
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/templates/default/hdp_hdfs-site.xml.erb]: Could this have a node attribute too? (100MB is a drop in the bucket for nodes with 20 and 40GB throughput.)

Comment (amithkanand) [inline cookbooks/bcpc-hadoop/templates/default/hdp_hdfs-site.xml.erb]: Yes, can be converted into a node attribute. I need to figure out a way to get max speed for network card to be able to calculate it based on a ratio.

Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/templates/default/hdp_hdfs-site.xml.erb]: It looks like `ethtool` supports general hardware but `mii-tool` does not support all the NICs I could find.

**Comment by cbaenziger** [comment]
@amithkanand Awaiting at least manual configuration of `dfs.datanode.balance.bandwidthPerSec` to an arbitrary value. I'm happy to just have an attribute, no need to make it a ratio of interface speed.

**Comment by cbaenziger** [comment]
The zabbix-agent removal also needs to strip references from:
* [keepalived-vip_change.erb](https://github.com/bloomberg/chef-bach/blob/be32e86ca5e961824c8b094e694bffc804a84103/cookbooks/bcpc/templates/default/keepalived-vip_change.erb#L19-L20)
* [zabbix-head.rb](https://github.com/bloomberg/chef-bach/blob/8afefe5dfab92333b7611005889f1d7f903f7d78/cookbooks/bcpc/recipes/zabbix-head.rb#L169-L175)

**Comment by cbaenziger** [comment]
Due to [graphite_to_zabbix.rb](https://github.com/bloomberg/chef-bach/blob/a278f13657b5caa4d6507ddf93036abfd2aa6657/cookbooks/bcpc-hadoop/recipes/graphite_to_zabbix.rb) we need zabbix_sender on head nodes. Thanks for dropping #140 from this for now.

**Pull request merged**